### PR TITLE
Simplified regex to pass tests

### PR DESCRIPTION
**Pull request by aaren-cordova**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -5,7 +5,9 @@ const isNullOrUndefined = require('./helpers/is-null-or-undefined');
 const VNodeFlags = require('./flags');
 
 function handleWhiteSpace(str) {
-	return str.replace(/[\s\t]{2,}|[\r\n]|[\s\t]+?=[\s\t]/g, '');
+	const twoOrMoreWhiteSpaces = /\s{2,}/g;
+	return str.replace(twoOrMoreWhiteSpaces, '');
```
Comment (lukeed) [inline lib/index.js]: Please remove the constant declaration. The expression isn't used more than once. :)

**Comment by trueadm** [comment]
@nightwolfz please can your code review too?

File: lib/index.js (hunk 1 above)
Comment (trueadm) [inline lib/index.js]: Well it can be hoisted out of this closure

Comment (lukeed) [inline lib/index.js]: Ok, I see. My b

**Comment by nightwolfz** [comment]
This regex leaves a single empty space at beginning. Is that the behavior we want ?

**Comment by lukeed** [comment]
IIRC, we want to leave a space at the end _only_ so that `Hello world, ` retains its trailing space.

**Comment by nightwolfz** [comment]
Some "failing" examples

```
const regex = /\s{2,}/g

`\nBlow your mind\n`.replace(regex, ' ');    // leaves new line at beginning/end
`\nBlow\n your\n mind\n`.replace(regex, ''); // removes new lines and spaces
`\t Blow\t your  mind`.replace(regex, '');  // removes all spaces
` \t Blow your mind \t`.replace(regex, '');  // doesn't leave space at the beginning and end
```

I think they should all give the same output

**Comment by aaren-cordova** [comment]
Awesome, I was relying solely on the tests to define the bounds of what is proper input and output text.  I'll update it to include the failing examples @nightwolfz mentioned.

**Comment by aaren-cordova** [comment]
Getting a few tests failing, but I'm not quite sure they line up with the above comment.
Tests like:
`<div class="tab-group">inlineText\t</div>` to equal `<div class="tab-group">inlineText</div>`

The above changes should now convert it to:
`<div class="tab-group">inlineText \n \r     \t</div>` to equal `<div class="tab-group">inlineText\t</div>`

So I may need update a few of the tests in inferno before I can move this one on through.

**Comment by trueadm** [comment]
I've added a fix (taken from the Mithril MSX plugin) that seems to work. Even if it is very long winded and much slower – it seems to work in all the test cases.

**Pull request closed without merging**